_posts/2023/2023-02-15-cobbler-inside-libvirt-1.md

```diff
@@ -0,0 +1,129 @@
+---
+layout: post
+title: Cobbler inside libvirt (1)
+author: Enno
+summary: Cobbler inside libvirt with a private libvirt network
+---
+
+The last blog post that was neither a roadmap update or a release announcement was in december of 2020. Thus it is time
+to give you guys a new guide!
+
+This time we will together explore how to use Cobbler to enable a libvirt internal network to boot over the network.
+The guide will assume you have a running host OS and libvirt installed and running already. As a guest OS I will use
+openSUSE Tumbleweed but any other guest OS should work equally good.
+
+In case you have libvirt not installed please consult your OS vendors resources to do so. A few examples:
+
+- [openSUSE](https://doc.opensuse.org/documentation/leap/virtualization/html/book-virtualization/part-virt-libvirt.html)
+- [Fedora](https://developer.fedoraproject.org/tools/virtualization/installing-libvirt-and-virt-install-on-fedora-linux.html)
+- [Ubuntu](https://ubuntu.com/server/docs/virtualization-libvirt)
+
+Note:
+
+> There will be a second part of this guide where we will utilize a network bridge to also boot real hardware with our
+> VM based Cobbler setup.
+
+Information:
+
+> Depending on your setup you may need to prefix the `virsh` and other commands with `sudo`.
+
+1. Create libvirt network for Cobbler (don't use the network `default`).
+   ```xml
+   <network>xml
+     <name>cobbler</name>
+     <uuid>6b7f4f36-7895-4b74-8be6-dfa4b2b84500</uuid>
+     <forward mode='nat'/>
+     <bridge name='virbr2' stp='on' delay='0'/>
+     <mac address='52:54:00:fb:72:57'/>
+     <domain name='cobbler.local'/>
+     <dns enable='no'/>
+     <ip family='ipv4' address='10.17.3.1' prefix='24'>
+     </ip>
+   </network>
+   ```
+2. Download the openSUSE Tumbleweed ISO and move it to your desired libvirt storage pool (below I show the default
+   location):
+   ```
+   wget https://download.opensuse.org/tumbleweed/iso/openSUSE-Tumbleweed-NET-x86_64-Current.iso
+   mv openSUSE-Tumbleweed-NET-x86_64-Current.iso /var/lib/libvirt/images
+   ```
+3. Setup the guest. An example for a guide can be found [in the openSUSE Documentation](https://doc.opensuse.org/documentation/leap/virtualization/html/book-virtualization/cha-kvm-inst.html#sec-libvirt-inst-vmm).
+4. Login into the guest either via SSH or if you are using `virt-manager` using the GUI.
+5. `zypper in cobbler` (atm 3.3.3)
+6. Shutdown the VM from the host
+   ```
+   virsh shutdown "<vm name>"
+   ```
+   or if you are logged into the guest use
+   ```
+   systemctl poweroff
+   ```
+7. Add a second network to the guest. Use the network defined in step 1.
```

Comment:
You should mention that the example network above cannot be reused 1:1.
You need to change the network name and the bridge device to use.
I think a sub sentence like "Use the network defined in step 1 as a reference" would suffice.